src/core/Rational.pm6

```diff
@@ -129,7 +129,7 @@ my role Rational[::NuT = Int, ::DeT = ::("NuT")] does Real {
                     $fract *= 1000000;
                 }
                 else {
-                    $precision = nqp::chars($!denominator.Str) + nqp::chars($whole.Str) + 1;
+                    $precision = nqp::chars($!denominator.Str) + 1;
                     $fract *= nqp::pow_I(10, nqp::decont($precision), Num, Int);
                 }
             }
```

Comment:
What's the difference between this conditional branch and the truthy branch? The code in both looks the same.

Reply:
After this pull request, they would be the same. There was some speculative code stringifying FatRats to a higher precision, so there are two code paths. If the higher precision FatRat stringification isn't going in core, this could be collapsed to a single code path. (and the longer I think about it, the more I lean that direction.)

I've been puttering with a module to do high precision stringification. It's slow, but only people who specifically want it have to pay the cost.

Reply:
> speculative code

It could just be a text comment for future programmers without any code and we'd be able to get rid of duplicate code and the now-useless typecheck in the conditional there as well.